internal/typescript_mock.bzl

```diff
@@ -11,4 +11,22 @@ mock_typescript_lib = rule(
   attrs = {
     "srcs": attr.label_list(allow_files = True),
   }
-)
\ No newline at end of file
+)
+
+def _mock_es6_typescript_lib(ctx):
+  transitive_es6_sources = []
+  for s in ctx.attr.srcs:
+    transitive_es6_sources += s.files.to_list()
```

Comment:
`to_list` and back to depset is a smell, try

```
transitive_es6_sources = depset()
for s in ctx.attr.srcs:
  transitive_es6_sources = depset(transitive=[s.files, transitive_es6_sources])
```

Reply:
donezo